parser/binary-parser_test.go

```diff
@@ -0,0 +1,117 @@
+package parser
+
+import (
+	"encoding/json"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+
+	tfjson "github.com/hashicorp/terraform-json"
+)
+
+// MockCommandExecutor mocks the command execution.
+type MockCommandExecutor struct {
+	Output     []byte
+	Err        error
+	ActualName string
+	ActualArgs []string
+}
+
+func (e *MockCommandExecutor) CombinedOutput(name string, args ...string) ([]byte, error) {
+	e.ActualName = name
+	e.ActualArgs = args
+	return e.Output, e.Err
+}
+
+func TestBinaryParser_Parse_Success(t *testing.T) {
+	// Prepare mock output
+	mockPlan := tfjson.Plan{
+		FormatVersion:    "0.1",
+		TerraformVersion: "1.0.0",
+	}
+	output, err := json.Marshal(mockPlan)
+	if err != nil {
+		t.Fatalf("Failed to marshal mock plan: %v", err)
+	}
+
+	// Create parser with mock executor
+	parser := BinaryParser{
+		fileName: "mock-file",
+		executor: &MockCommandExecutor{
+			Output: output,
+			Err:    nil,
+		},
+	}
+
+	// Call Parse
+	parsedPlan, err := parser.Parse()
+
+	//assertions
+	assert.NoError(t, err, "Expected no error")
+	assert.Equal(t, mockPlan, parsedPlan, "Parsed plan does not match expected plan")
```

Comment:
we can skip the 4th argument. assert itself will print clearly if there is failure.

Reply:
Sure.